x-pack/plugins/lens/public/app_plugin/mounter.tsx

```diff
@@ -132,6 +134,11 @@ export async function mountApp(
         onAppLeave={params.onAppLeave}
         setHeaderActionMenu={params.setHeaderActionMenu}
         history={routeProps.history}
+        initialContext={
+          historyLocationState && historyLocationState.type === ACTION_VISUALIZE_LENS_FIELD
+            ? historyLocationState.payload
```

Comment:
This is okay, but I was almost expecting the entire `historyLocationState` to be passed in here so that the app is making fewer decisions. No need to change it unless you're planning on making other changes.